controller-server/src/main/java/com/yahoo/vespa/hosted/controller/deployment/DeploymentTrigger.java

```diff
@@ -277,7 +277,8 @@ private ApplicationVersion targetApplication(Application application, Change cha
                     : application.deploymentSpec().steps();
             List<Step> productionSteps = steps.stream().filter(step -> step.deploysTo(prod) || step.zones().isEmpty()).collect(toList());
 
-            Optional<Instant> completedAt = Optional.of(Instant.EPOCH); // Delays before first production job are ignored.
+            Optional<Instant> completedAt = application.deploymentJobs().statusOf(stagingTest)
+                                                       .flatMap(JobStatus::lastSuccess).map(JobRun::at);
```

Comment:
Delay will now be from staging for first production job, and from previous production jobs for later production jobs. I'll see if I can make it count down from the latter of the two, for later production jobs.